.github/workflows/canary.yaml

```diff
@@ -0,0 +1,76 @@
+name: Canary build
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+env:
+  GO_VERSION: "1.18"
+
+jobs:
+  Canary:
+    name: Canary
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup Go 
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Cache Go modules
+        uses: actions/cache@v3.0.2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Release
+        uses: goreleaser/goreleaser-action@v3
+        with:
+          version: v1.4.1
+          args: build -f=goreleaser-canary.yml --snapshot --rm-dist
+
+      - name: Copy contribution files for upload
+        run: |
+          for dir in dist/*/; do
+            mkdir $dir/contrib
+            cp {README.md,LICENSE} $dir
+            cp contrib/*.tpl $dir/contrib
+          done
+
+      # Upload artifacts 
+      - name: Upload artifacts (trivy_Linux-64bit)
+        uses: actions/upload-artifact@v3
+        with:
+          name: trivy_Linux-64bit
+          path: dist/trivy_canary_build_linux_amd64
+          if-no-files-found: error
+      
+      - name: Upload artifacts (trivy_Linux-ARM64)
+        uses: actions/upload-artifact@v3
+        with:
+          name: trivy_Linux-ARM64
+          path: dist/trivy_canary_build_linux_arm64
+          if-no-files-found: error
+
+      - name: Upload artifacts (trivy_macOS-64bit)
+        uses: actions/upload-artifact@v3
+        with:
+          name: trivy_macOS-64bit
+          path: dist/trivy_canary_build_darwin_amd64
+          if-no-files-found: error
+
+      - name: Upload artifacts (trivy_macOS-ARM64)
+        uses: actions/upload-artifact@v3
+        with:
+          name: trivy_macOS-ARM64
+          path: dist/trivy_canary_build_darwin_arm64
+          if-no-files-found: error
```

Comment:
It's been a while, so I forgot the reason why we can't use multiple paths. Could you explain it again?
https://github.com/actions/upload-artifact#upload-using-multiple-paths-and-exclusions

Reply:
all files will be saved in 1 archive when we use multiple paths for one step `actions/upload-artifact`.
In this case, we will save 4 Trivy bins in 1 archive.

Reply:
Oh, yes, I just remember it. Thanks for explaining it again!